File: kangaroo_app_sdk_android/android/src/main/kotlin/com/kangaroo/flutterplugin/android/base/PluginChannelHandler.kt
package com.kangaroo.flutterplugin.android.base

import io.flutter.plugin.common.EventChannel
import io.flutter.plugin.common.MethodCall

interface PluginChannelHandler {
    val methodChannel: String
    val eventChannel: String
    fun onMethodCall(call: MethodCall): Unit?
    fun getStreamHandler(): EventChannel.StreamHandler
}

File: kangaroo_app_sdk_macos/macos/Classes/base/PluginChannelHandler.swift
import Foundation
import FlutterMacOS

protocol PluginChannelHandler {
    var methodChannel: String { get set}
    var eventChannel: String { get set}

    func onMethodCall(call: FlutterMethodCall) async -> Any?

    func getStreamHandler() -> (FlutterStreamHandler & NSObjectProtocol)?
}
